### ISSUE TITLE
support for  X-Real-IP or X-Forwarded-For

### ISSUE THREAD

**Issue by brunoqc** [iron/iron#481]
Maybe a helper fonction or something to get the real IP when behind a proxy.

**Comment by untitaker** [comment]
That should be a BeforeMiddleware and probably also its own crate. I have one here: https://github.com/untitaker/mysteryshack/blob/master/src/mysteryshack/web/utils.rs#L22

**Comment by untitaker** [comment]
Also this thing is possibly incomplete, e.g. it only modifies the request URI and not the `Host` header too.

**Comment by Hoverbear** [comment]
This is an often used feature! 

If you're interested in working on this @brunoqc we can definitely support you. :) I'm sure people would use it.

**Comment by llompartg** [comment]
Hi

Is this still open pending?

**Comment by untitaker** [comment]
Yes it appears so, though in the meantime multiple packages for this may exist. If one could compile a list, we can close this.

On 10 May 2017 00:33:10 GMT+02:00, Guillem Llompart <notifications@github.com> wrote:
>Hi
>
>Is this still open pending?
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/iron/iron/issues/481#issuecomment-300320962

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by sunng87** [comment]
I believe this should be done in an iron ecosystem crate, instead of inside iron itself. As a micro server library, iron has a beautiful, extensiable design to do this externally. We should keep the core compact.

**Comment by kczimm** [comment]
Is this still a relevant issue?
